### PR TITLE
Prepare build for Scala3

### PR DESCRIPTION
**Pull request by octonato**
This is a initial setup to gradually start to compile with Scala3. 

The next step is to upgrade all the dependencies to their Scala3 releases. That can be challenging. I already run into an issue, for example spec2-mock was not published for Scala3.

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -223,6 +223,22 @@ jobs:
...
+      java: 11
+      scala: 3.1.2
+      # doing nothing for now, but as we progress we should add here the modules that are ready for compilation
+      # with Scala 3
+      # sbt "-Dscala.version=$SCALA_VERSION
+      #  Play / compile
+      #  Play / Test / compile
+      #  "
+      cmd: >-
+        sbt "-Dscala.version=$SCALA_VERSION"
```
Comment (octonato) [inline .github/workflows/build-test.yml:239]: Doing nothing for the moment and that's intentional. Once we solve the dependencies we can start to hack the modules and add then here when ready. 

The first one should look like:

```yml
  scala3-compilation:
    name: Build with Scala3
    uses: playframework/.github/.github/workflows/sbt.yml@v1
    with:
      java: 11
      scala: 3.1.2
      cmd: >-
        sbt " -Dscala.version=$SCALA_VERSION
             Play / Test / compile
             "
```

**Comment by octonato** [comment]
@mkurz, I see that you have already run into the specs2-mock issue in play-ws. 
https://github.com/playframework/play-ws/pull/668 

We will probably need to use CrossVersion.for3Use2_13 in a few places.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -148,6 +148,21 @@ jobs:
...
```
Comment (ihostage) [inline .github/workflows/build-test.yml:164]: For `cmd.yml@v2` should be
```suggestion
        sbt "-Dscala.version=$MATRIX_SCALA"
```

Comment (mkurz) [reply .github/workflows/build-test.yml]: Thanks, force pushed again.

**Comment by octonato** [comment]
The problem is that the sbt command never returns. The way I did make it start the sbt console and it stays there, forever. We need to add a dummy command to make it run and quit.

**Comment by octonato** [comment]
I'm now calling `sbt scalaVersion`. 

Let's see what the build says.

**Comment by octonato** [comment]
hmm, build still hands forever

**Comment by octonato** [comment]
I think this is ready for review. 

I got some trouble with the CI syntax to use multi-line commands, but that's a details. For now I made it one-liner. We see later how we want to add the modules. 

Once this is merged, I think the next step will be to update the dependencies to have them all using scala 3.

**Review by mkurz — APPROVED** [review]

**Comment by mkurz** [comment]
> @mkurz, I see that you have already run into the specs2-mock issue in play-ws. [playframework/play-ws#668](https://github.com/playframework/play-ws/pull/668)
> 
> We will probably need to use CrossVersion.for3Use2_13 in a few places.

Yeah, the author of specs2 [recommends switching to ScalaMock](https://github.com/etorreborre/specs2/issues/1079#issuecomment-1092863601), but that doesn't have Scala 3 artifacts yet as well: https://github.com/paulbutcher/ScalaMock/issues/429
BTW: Here are some notes for the specs 4.15 release: https://github.com/etorreborre/specs2/releases/tag/SPECS2-4.15.0

**Pull request merged**